_posts/2020-11-01-batch-processing-stop-job-step.adoc

```diff
@@ -0,0 +1,229 @@
+`stop()` method that a batchlet class must implement.
+
+In the example batchlet class below, once it receives a stop request in its `stop()` method, it sets the `toStop` flag
+to `true`. Its `process()` method periodically checks this flag to determine if it needs to stop processing.
+
+Note that bathlet `stop()` method is called asynchronously while the batchlet `process()` method is running in a
+different thread. And the batchlet class should be implemented to properly handle concurrency.
+
+[source,java]
+----
+@Named
+public class Batchlet1 implements Batchlet {
+    private final AtomicBoolean toStop = new AtomicBoolean();
+
+    @Override
+    public String process() throws Exception {
+        String exitStatus = "BATCHLET1_COMPLETED";
+        while (true) {
+            if (toStop.get()) {
+                exitStatus = "BATCHLET1_STOPPED";
+                break;
+            }
+            // perform batchlet task
+            // Thread.sleep(5000);
+        }
+        return exitStatus;
+    }
+
+    @Override
+    public void stop() throws Exception {
+        toStop.set(true);
+    }
+}
+----
+
+After the batch application is deployed to WildFly, you can start and stop a job execution in WildFly CLI:
+
+[source,text]
+----
+# To start a new job execution
+#
+/deployment=numbers-chunk.war/subsystem=batch-jberet:start-job(job-xml-name=numbers)
+{
+    "outcome" => "success",
+    "result" => 1L
+}
+
+# To stop the running job execution
+#
+/deployment=numbers-chunk.war/subsystem=batch-jberet:stop-job(execution-id=1)
+{"outcome" => "success"}
+----
+
+The above `stop-job` CLI operation calls `JobOperator.stop(jobExecutionId)` behind the scene, and eventually the
+batchlet `stop()` method is called to receive the stop request.
+
+The following CLI commands checks the status of the stopped job execution, and a stopped job execution can be
+restarted from where it left off.
+
+[source,text]
+----
+# To check the status of the stopped job execution:
+#
+/deployment=numbers-chunk.war/subsystem=batch-jberet/job=numbers/execution=1:read-resource(include-runtime, recursive)
+{
+    "outcome" => "success",
+    "result" => {
+        "batch-status" => "STOPPED",
+        "create-time" => "2020-10-29T19:33:13.843-0400",
+        "end-time" => "2020-10-29T19:33:30.258-0400",
+        "exit-status" => "STOPPED",
+        "instance-id" => 1L,
+        "last-updated-time" => "2020-10-29T19:33:30.258-0400",
+        "start-time" => "2020-10-29T19:33:13.853-0400"
+    }
+}
+
+# To restart the previously stopped job execution:
+#
+/deployment=numbers-chunk.war/subsystem=batch-jberet:restart-job(execution-id=3)
+{
+    "outcome" => "success",
+    "result" => 4L
+}
+----
+
+You can also perform all the above operations in WildFly Management Console. For example, the following
+screenshot shows the UI to stop a job execution:
+
+image::jberet/jberet-stop-job.png[Stop Batch Job Execution]
```

Comment:
Something is wrong with image resolution in general. I'm going to have a look at this as other posts are also missing images.

Reply:
I found the issue with this and filed https://github.com/wildfly/wildfly.org/issues/209. Nothing to change here as I'll send up a global fix.

Reply:
#209 has been resolved, so the image should work now.